src/Datagrid/Datagrid.php

```diff
@@ -151,7 +151,9 @@ static function ($value) {
         foreach ($this->getFilters() as $name => $filter) {
             $this->values[$name] = isset($this->values[$name]) ? $this->values[$name] : null;
             $filterFormName = $filter->getFormName();
-            if (isset($this->values[$filterFormName]['value']) && '' !== $this->values[$filterFormName]['value']) {
+            if (isset($this->values[$filterFormName]['type'], $this->values[$filterFormName]['value']) &&
```

Comment:
This method is already too long. Can you please extract a method for this?

Bonus points if you can extract other parts too.

Reply:
I did it